module/zfs/zfs_vnops.c

```diff
@@ -4636,13 +4645,22 @@ zfs_dirty_inode(struct inode *ip, int flags)
 	}
 #endif
 
+top:
 	tx = dmu_tx_create(zfsvfs->z_os);
 
 	dmu_tx_hold_sa(tx, zp->z_sa_hdl, B_FALSE);
 	zfs_sa_upgrade_txholds(tx, zp);
 
-	error = dmu_tx_assign(tx, TXG_WAIT);
+	boolean_t waited = B_FALSE;
```

Comment:
Please move this to the beginning of the function

Reply:
Note that this is not just a stylistic suggestion; if we `goto top`, we want `waited` to be `TRUE`